cg/meta/archive/archive.py

```diff
@@ -53,7 +55,29 @@ def __init__(
 
     def archive_files(self, files: List[FileAndSample], archive_location: ArchiveLocations) -> int:
         archive_handler: ArchiveHandler = ARCHIVE_HANDLERS[archive_location](self.data_flow_config)
-        return archive_handler.archive_folders(files_and_samples=files)
+        return archive_handler.archive_files(files_and_samples=files)
+
+    def archive_all_non_archived_spring_files(
+        self, spring_file_count_limit: int = DEFAULT_SPRING_ARCHIVE_COUNT
+    ) -> None:
+        """Archives all non archived spring files."""
+
+        files_to_archive: List[File] = self.housekeeper_api.get_all_non_archived_spring_files()[
```

Comment:
Here we always query for all non archived files and pass them through. Would having a function that just fetches a limited number be beneficial here to avoid passing too many unused objects?
something like.

```Python 
if spring_file_count_limit:
 files_to_archive: List[File] = self.housekeeper_api.get_n_non_archived_spring_files(n = spring_file_count_limit)
 ```

Reply:
Yes, I do think that would be better. I think I will keep it as is for now, though, until such functionality has been added in HouseKeeper.